apps/logging/serializers.py

```diff
@@ -360,18 +374,37 @@ def get_type(self):
         return 'SLSx_userinfo'
 
     def to_dict(self):
-        event_dict = json.loads(self.resp.text)
+
+        event_dict = {}
+        json_exception = {}
+
+        try:
+            event_dict = json.loads(self.resp.text)
+        except json.decoder.JSONDecodeError:
+            json_exception = {
+                "message": "JSONDecodeError thrown when parsing response text."
+            }
+
         event_dict.update(super().to_dict().copy())
+
         resp_dict = {
-            "type": event_dict['type'],
-            "uuid": event_dict['uuid'],
-            "path": event_dict['path'],
-            "sub": event_dict['data']['user']['id'],
-            "code": event_dict['code'],
-            "size": event_dict['size'],
-            "start_time": event_dict['start_time'],
-            "elapsed": event_dict['elapsed'],
+            "type": event_dict.get('type'),
+            "uuid": event_dict.get('uuid'),
+            "path": event_dict.get('path'),
+            "sub": event_dict.get('data').get('user').get('id')
+            if event_dict.get('data') is not None
+            and event_dict.get('data').get('user') is not None
+            and event_dict.get('data').get('user').get('id') is not None
+            else None,
```

Comment:
@JFU-GIT @njdister Was just testing this code out and I think it will work for multi-level dictionary key existence cases like this. In testing, this returns `None` if any of the keys in the levels don't exist.

The code change would be:

```suggestion
            "sub": event_dict.get('data', {}).get('user', {}).get('id'),
```